pyhub/mcptools/google/sheets/client_async.py

```
@@ -0,0 +1,565 @@
+                raise GoogleSheetsError(f"Permission denied: {e}")
+            elif ErrorPatterns.matches_pattern(error_text, ErrorPatterns.QUOTA_KEYWORDS):
+                raise RateLimitError(f"API quota exceeded: {e}")
+            elif ErrorPatterns.matches_pattern(error_text, ErrorPatterns.NOT_FOUND_KEYWORDS):
+                raise SpreadsheetNotFoundError(f"Resource not found: {e}")
+            elif ErrorPatterns.matches_pattern(error_text, ErrorPatterns.AUTHENTICATION_KEYWORDS):
+                raise AuthenticationError(f"Authentication failed: {e}")
+            else:
+                raise GoogleSheetsError(f"Google API error: {e}")
+        else:
+            logger.error(f"Unexpected error: {type(e).__name__}: {e}")
+            raise GoogleSheetsError(f"Operation failed: {e}")
```

Pylint (3.3.7)

[refactor] Unnecessary "elif" after "raise", remove the leading "el" from "elif"

(R1720)